tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -355,6 +368,26 @@ static PyObject * THPVariable_from_numpy(PyObject* module, PyObject* arg)
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject * THPVariable__promote_types(PyObject* self, PyObject* args, PyObject* kwargs)
```

Comment:
What's going on here?

Reply:
@prasunanand can you comment here?

Reply:
I have plugged in `__torch__function__` . Here `r.has_torch_function` checks if  parsed signatures could find an overloaded Tensor-like argument.

And if `__torch__function__` is found, we return the control to overriding function.

Reply:
I guess what I am surprised about, is previously we didn't have any Python binding for `_promote_types`. Now there is one. How come this diff caused us to need this binding?

Reply:
Bumping this question

Reply:
I've pinged @prasunanand offline about this, but I think that these bindings aren't actually needed and have deleted them. I'm happy to restore them if Prasun corrects me.